docs/source/vision_tasks.rst

```diff
@@ -0,0 +1,88 @@
+Vision Tasks
+============
+Here we present common tasks used in computer vision.
+
+------------------
+
+Self-supervised learning
+------------------------
+Self-supervised learning in computer vision uses a pretext task to extract learning signal.
+Here is a collection of common tasks
+
+AMDIMContrastiveTask
+^^^^^^^^^^^^^^^^^^^^
+Implementation modified from the `original repo <https://github.com/Philip-Bachman/amdim-public>`_.
+This is the contrastive task from AMDIM (`Philip Bachman, R Devon Hjelm, William Buchwalter <https://arxiv.org/abs/1906.00910>`_).
+
+In this task, we take in two sets of feature maps
+(in this case from a positive $x^+$ and anchor example $x^a$) $M^a = \{m_1, m_2, ..., m_i\}, M^+ = \{m_1, m_2, ..., m_j\}$.
+
+The task compares feature maps across spatial locations within the network. Here are the supported possibilities:
+
+The `1:1` task:
+
+Corresponds to using the feature maps $m_{i} \in M^a$ vs $m_{j} \in M^+$ in the NCE Loss $\math{cal}(m_i, m_j)$
+
+.. code-block:: python
+
+    xa, x+ = augmentations(x), augmentations(x)
+    ra = model(xa)
+    r+ = model(x+)
```

Comment:
is this correct `r+`?

Reply:
rpos?

Reply:
it’s a positive, anchor and negative representation

Reply:
yeas but `+` is not allowed in var name... so `r_pos` ?

Reply:
probably not too important in code but in examples rather longer names so user can easier follow...

Reply:
as this example is no valid code lol